### PR TITLE
Flowsheet documentation for RO w/ ERD

### PR DESCRIPTION
**Pull request by kurbansitterley**
## Fixes/Resolves:

RO w/ ERD flowsheet documentation for #1219 

## Summary/Motivation:

Add flowsheet documentation for ``RO_with_energy_recovery.py``

## Changes proposed in this PR:
- add doc .rst file
- add figures for each ERD configuration

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: docs/technical_reference/flowsheets/RO_with_energy_recovery.rst [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+Reverse osmosis model
+    * :doc:`/technical_reference/unit_models/reverse_osmosis_0D`
+Pressure exchanger model
+    * :doc:`/technical_reference/unit_models/pressure_exchanger`
+IDAES Product blocks
+    * :doc:`idaes:reference_guides/model_libraries/generic/unit_models/product`
+IDAES Separator blocks
+    * :doc:`idaes:reference_guides/model_libraries/generic/unit_models/separator`
+IDAES Mixer blocks
+    * :doc:`idaes:reference_guides/model_libraries/generic/unit_models/mixer`
```
Comment (MarcusHolly) [inline docs/technical_reference/flowsheets/RO_with_energy_recovery.rst]: Similar to the other PR, I think these should be grouped into sections

File: docs/technical_reference/flowsheets/RO_with_energy_recovery.rst [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+* Initialize entire system with ``initialize_system()``:
+
+    After the model is built and specified, this function will initialize the feed, RO, and costing blocks.
+    Depending on the ``erd_type`` different helper functions are used to initialize the relevant 
+    ERD-related unit models.    
+
+* Optionally optimize the system with ``optimize_set_up()``:
+
+    The model is fully defined and can be solved with the provided ``solve()`` function after the user passes 
+    the model through ``initialize_system()``. This function is provided as a demostration of how the system could be
```
Comment (MarcusHolly) [inline docs/technical_reference/flowsheets/RO_with_energy_recovery.rst]:
```suggestion
    the model through ``initialize_system()``. This function is provided as a demonstration of how the system could be
```

File: docs/technical_reference/flowsheets/RO_with_energy_recovery.rst [hunk 3]
```diff
@@ -0,0 +1,162 @@
+Reverse Osmosis with Energy Recovery
+====================================
+
+Introduction
+------------
+
+This flowsheet represents a reverse osmosis (RO) with energy recovery device (ERD). 
+The flowsheet is similar to the desalination block of the full-scale seawater RO treatment facility flowsheet,
+but includes an demonstration for optimization of the process. The user can select between pressure exchanger ERD,
```
Comment (MarcusHolly) [inline docs/technical_reference/flowsheets/RO_with_energy_recovery.rst]:
```suggestion
but includes a demonstration for optimization of the process. The user can select between pressure exchanger ERD, 
```

File: docs/technical_reference/flowsheets/RO_with_energy_recovery.rst [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+Code Documentation
+------------------
+
+* :mod:`watertap.examples.flowsheets.RO_with_energy_recovery`
+
+References
+----------
+
+References for each component and model used in this flowsheet can be found using the links provided in the Implementation section of this documentation.
```
Comment (MarcusHolly) [inline docs/technical_reference/flowsheets/RO_with_energy_recovery.rst]: I think we can just delete this section. References should be reserved for literature references.

**Review by MarcusHolly — APPROVED** [review]
2 minor things to address,  but otherwise LGTM

**Review by savannahsakhai — APPROVED** [review]
LGTM

**Comment by lbianchi-lbl** [comment]
@kurbansitterley @adam-a-a FYI here's how I solved the merge conflict in the index page, let me know if something doesn't look right:

**Pull request merged**